recipes/Myst/myst_prepare.py

```diff
@@ -0,0 +1,560 @@
+
+"""
+Myst data preparation (SpeechBrain-style) — silence detection removed.
+
+This mirrors the API of `librispeech_prepare.py` while adding zero-shot
+WER filtering across all splits (train/valid/test).
+
+Outputs CSVs with columns:
+    ID,duration,wav,spk_id,wrd
+
+Expected layout per split directory:
+    <data_folder>/<split>/**/<audio>.(wav|flac|mp3|m4a|ogg)
+    with required sidecar transcripts: <audio>.trn
+
+Authors: Thomas Rolland 2025
+"""
+
+import csv
+import functools
+import os
+import random
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional, Sequence, Tuple
+from tqdm import tqdm
+import re
+
+
+# Optional dependencies used by LibriSpeech prep
+try:
+    from speechbrain.dataio.dataio import (
+        load_pkl,
+        merge_csvs,
+        read_audio_info,
+        save_pkl,
+    )
+    from speechbrain.utils.data_utils import get_all_files
+    from speechbrain.utils.logger import get_logger
+    from speechbrain.utils.parallel import parallel_map
+except Exception:  # graceful fallback if SpeechBrain is not present
+    load_pkl = None
+    merge_csvs = None
+    read_audio_info = None
+    save_pkl = None
+    parallel_map = None
+    def get_all_files(folder, match_and=None, match_or=None, exclude_or=None):
+        out = []
+        for root, _dirs, files in os.walk(folder):
+            for fn in files:
+                path = os.path.join(root, fn)
+                if match_or and not any(path.endswith(m) for m in match_or):
+                    continue
+                if match_and and not all(m in path for m in match_and):
+                    continue
+                if exclude_or and any(ex in path for ex in exclude_or):
+                    continue
+                out.append(path)
+        return out
+    class _Logger:
+        def info(self, *a, **k): print(*a)
+        def warning(self, *a, **k): print(*a)
+    def get_logger(name): return _Logger()
+
+logger = get_logger(__name__)
+OPT_FILE = "opt_myst_prepare.pkl"
+
+# Audio / text utilities
+AUDIO_EXTS = (".wav", ".flac", ".mp3", ".m4a", ".ogg")
+try:
+    from whisper_normalizer.english import EnglishTextNormalizer  # type: ignore
+except Exception:
+    EnglishTextNormalizer = None
```

Comment:
you can probably pass the normalizer directly in the input main function as the normalizer is accessible within the tokenizer object of whisper